File: 프로그래머스 풀이/lv1_모의고사.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    int man1[] = {1, 2, 3, 4, 5};
    int man2[] = {2, 1, 2, 3, 2, 4, 2, 5};
    int man3[] = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    int sum[3] = {0,0,0};
    
    int i, s;
    for (i = 0; i < answers.size(); i++) {
        if ( man1[i % 5] == answers[i] )
            sum[0]++;
        if ( man2[i % 8] == answers[i] )
            sum[1]++;
        if ( man3[i % 10] == answers[i] )
            sum[2]++;
    }
    answer.push_back(0);    // push idx
    for (i = 1; i < 3; i++) {
        s = sum[i];
        if ( sum[answer[0]] < s ) {
            answer.clear();
            answer.push_back(i);
        }
        else if ( sum[answer[0]] == s ){
            answer.push_back(i);   
        }
    }
    for (i = 0; i < answer.size(); i++)
        answer[i]++;
    return answer;
}

File: 프로그래머스 풀이/lv1_모의고사.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] answers) {
        int[] answer;
        int[][] man = {{1,2,3,4,5},{2,1,2,3,2,4,2,5},{3,3,1,1,2,2,4,4,5,5}};
        int[] sum = {0,0,0};
        int[] t = {0,0,0};

        for (int i = 0; i < 3; i++) 
            sum[i] = compareAnswer(answers, man[i]);
        
        int i = 1, idx = 1;
        int max = sum[0];
        t[0] = 1;
        while (i < sum.length) {
            if (max < sum[i]) {
                max = sum[i];
                t[0] = i+1;
                idx = 1;
            }
            else if (max == sum[i]) {
                t[idx] = i+1;
                idx++;
            }
            i++;
        }
        
        answer = new int[idx];
        for (i = 0; i < idx; i++) {
            answer[i] = t[i];
        }
        Arrays.sort(answer);
        
        return answer;
    }
    
    public int compareAnswer(int[] answers, int[] man) {
        int size = man.length;
        int sum = 0;
        for(int i = 0, a = 0; a < answers.length; i++, a++){
            if(i == size)   i = 0;
            
            if(man[i] == answers[a] )  sum = sum + 1;
        }
        return sum;
    }
}